### DISCUSSION TITLE
please help me for unable to detect my gpu!

### DISCUSSION THREAD

**Discussion by heyingu** [ROCm/ROCm#3643] [Q&A] [answered]
My GPU is 6750gre, linux system chosen is ubuntu22.04.4, ROCM version tried the latest version and 5.4.2, pytorch also selected the corresponding version. But whenever I use the rocm-smi command the gpu is not detected and I need to run the sudo modprobe amdgpu command before each use, otherwise the file cannot be found. I'm trying to figure out what's wrong, and I've gone back and forth and retried 3 times, taking two or three days. I hope some kind soul can clear up the confusion for me, thanks.

**Comment by taylding-amd** [comment, marked as answer]
Hi @heyingu, here are some steps and considerations that might help you troubleshoot and resolve the issue:
1. Check Permissions
The most likely cause of not detecting GPU is not having the necessary permissions to access GPU. You may need to add your user to the render and video group. You will have to reboot the system for the it to apply. 
`sudo usermod -a -G render,video $LOGNAME`
1. Check Kernel Version and Other System requirements
Ensure that your kernel version is compatible with the ROCm version you are using. ROCm has specific kernel requirements, and using an unsupported kernel can lead to issues. [System requirements (Linux)](https://rocm.docs.amd.com/projects/install-on-linux/en/latest/reference/system-requirements.html)
3.  Check AMDGPU Driver Installation
Ensure that the AMDGPU driver is installed correctly. You can check if the driver is loaded by `lsmod | grep amdgpu`
4.  Check for Secure Boot
If your system has Secure Boot enabled, it may prevent the AMDGPU driver from loading. You can either disable Secure Boot in your BIOS settings or sign the driver modules.
5. Check ROCm Installation
Make sure that the newest version of ROCm is installed correctly. You can verify the installation by checking the ROCm version: /opt/rocm/bin/rocm-smi
6. Review Logs
Check system logs for any errors related to the GPU or ROCm. `dmesg | grep amdgpu`
